openmc/mgxs/mgxs.py

```diff
@@ -4929,24 +5354,41 @@ def get_xs(self, groups='all', subdomains='all', nuclides='all',
         xs = np.squeeze(xs, axis=len(xs.shape) - 1)
         xs = np.nan_to_num(xs)
 
-        # Reshape tally data array with separate axes for domain and energy
         if groups == 'all':
             num_groups = self.num_groups
         else:
             num_groups = len(groups)
 
-        num_subdomains = int(xs.shape[0] / num_groups)
-        new_shape = (num_subdomains, num_groups) + xs.shape[1:]
+        # Reshape tally data array with separate axes for domain and energy
+        # Accomodate the polar and azimuthal bins if needed
+        num_subdomains = int(xs.shape[0] / (num_groups * self.num_polar *
+                                            self.num_azimuthal))
+        if self.num_polar > 1 or self.num_azimuthal > 1:
+            new_shape = (self.num_polar, self.num_azimuthal, num_subdomains,
+                         num_groups) + xs.shape[1:]
+        else:
+            new_shape = (num_subdomains, num_groups) + xs.shape[1:]
         xs = np.reshape(xs, new_shape)
 
         # Reverse data if user requested increasing energy groups since
         # tally data is stored in order of increasing energies
         if order_groups == 'increasing':
-            xs = xs[:, ::-1, :]
+            xs = xs[..., ::-1, :]
 
         if squeeze:
-            xs = np.squeeze(xs)
-            xs = np.atleast_1d(xs)
+            # We want to squeeze out everything but the polar, azimuthal,
+            # and energy group data.
+            if self.num_polar > 1 or self.num_azimuthal > 1:
+                dont_squeeze = (0, 1, 3)
+            else:
+                dont_squeeze = (1,)
+            # Squeeze will return a ValueError if the axis has a size
```

Comment:
I really like this suggestion, and I totally agree with the convoluted nature of this module (though im sure most is by necessity).  

I will do this particular fix in this PR, but I also want to take a stab at combining/making methods more generic in a follow-on PR